R/boundary_designs.R

```diff
@@ -0,0 +1,145 @@
+#' Boundary designs
+#'
+#' Compute boundary designs. These can be used for optimization and are
+#' implemented by default in \link{minimize}.
+#'
+#' @param objective An object of class \code{\link{UnconditionalScore-class}} to be minimized
+#' @param subject_to A \code{\link{ConstraintsCollection-class}} defining the constraints
+#' @param initial_design The initial design
+#' @param ... optimal arguments
+#'
+#' @rdname boundary-designs
+#' @export
+setGeneric("get_lower_boundary_design",
+           function(objective, subject_to, initial_design, ...) standardGeneric("get_lower_boundary_design"))
```

Comment:
why do you need to optimize anything to just get the boundary design?
this function should only depend on the initial design, right? Your solution still depends on a arbitrary input to os design. You will never get a problem agnostic heuristic. You could make c1e an additional parameter  for get_lower_boundary design with default 1 or so, don't overcomplicate it ;)